sdk/search/Azure.Search.Documents/src/Indexes/Models/BlobExtractionMode.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Search.Documents.Indexes.Models
+{
+    /// <summary>
+    /// Defines which parts of a blob will be indexed by the blob storage indexer.
+    /// See <see href="https://docs.microsoft.com/azure/search/search-howto-indexing-azure-blob-storage" /> for details.
+    /// </summary>
+    public readonly partial struct BlobExtractionMode : IEquatable<BlobExtractionMode>
+    {
+        private readonly string _value;
+
+        /// <summary> Determines if two <see cref="BlobExtractionMode"/> values are the same. </summary>
+        public BlobExtractionMode(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string StorageMetadataValue = "storageMetadata";
+        private const string AllMetadataValue = "allMetadata";
+        private const string ContentAndMetadataValue = "contentAndMetadata";
+
+        /// <summary>
+        /// Specifies that only the standard blob properties and user-specified metadata will be indexed.
+        /// See <see href="https://docs.microsoft.com/azure/storage/storage-properties-metadata" /> for details.
+        /// </summary>
+        public static BlobExtractionMode StorageMetadata { get; } = new BlobExtractionMode(StorageMetadataValue);
+
+        /// <summary>
+        /// Specifies that storage metadata and the content-type specific metadata extracted from the blob content will be indexed.
+        /// See <see href="https://docs.microsoft.com/azure/search/search-howto-indexing-azure-blob-storage#content-type-specific-metadata-properties" /> for details.
+        /// </summary>
+        public static BlobExtractionMode AllMetadata { get; } = new BlobExtractionMode(AllMetadataValue);
+
+        /// <summary>
+        /// Specifies that all metadata and textual content extracted from the blob will be indexed. This is the default value.
```

Comment:
Does that mean Metadata + Properties when you say all metadata?

Reply:
Same as above: extensible enum names should match enum values.

Reply:
Sounds like we should have a naming conversation and make it consistent everywhere.